sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/common/ImmutableLongArray.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.metrics.common;
+
+import java.util.Arrays;
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.Immutable;
+
+@Immutable
+public class ImmutableLongArray {
```

Comment:
Same comment as for the Double version, I would prefer to not expose this publicly.